src/hnsw/utils.c

```diff
@@ -33,7 +36,77 @@ void PopulateUsearchOpts(Relation index, usearch_init_options_t *opts)
     opts->connectivity = HnswGetM(index);
     opts->expansion_add = HnswGetEfConstruction(index);
     opts->expansion_search = HnswGetEf(index);
-    opts->metric_kind = usearch_metric_l2sq_k;
+    opts->metric_kind = HnswGetMetricKind(index);
     opts->metric = NULL;
     opts->quantization = usearch_scalar_f32_k;
 }
+
+int IsInsideQuotes(const char *query, int op_start, int op_end)
+{
+    if(op_start == 0) return 0;
+
+    int    single_quote_start = -1;
+    size_t offset = 0;
+
+    while(strcmp(query + offset, "\0") != 0) {
+        char s = *(query + offset);
+        if(s == '\'') {
+            // if current char is single quote
+            if(single_quote_start == -1) {
+                // if we didn't encounter any unclosed single quote before
+                // keep the start offset
+                single_quote_start = offset;
+            } else {
+                // if we already have open single quote
+                // check if our operator is inside quotes
+                // return 1
+                if(single_quote_start < op_start && offset > op_end) {
+                    return 1;
+                }
+                single_quote_start = -1;
+            }
+        }
+        offset += 1;
+    }
+
+    return 0;
+}
+
+int CheckOperatorUsage(const char *query)
+{
+    const char *pattern = "(<->)";
+    const char *orderby_pattern = "order by";
+    regex_t     regex;
+    regex_t     regex2;
+    regmatch_t  matches[ 1 ];
+    int         reti;
+    int         reti2;
+    int         status = 0;
+
+    reti = regcomp(&regex, pattern, REG_EXTENDED | REG_ICASE);
+    reti2 = regcomp(&regex2, orderby_pattern, REG_EXTENDED | REG_ICASE);
+    if(reti || reti2) {
+        elog(ERROR, "Could not compile regex");
```

Comment:
could this ever fail? Do `assert`, if not.

Reply:
if it could happen, make sure to call `regfree` before erroring

Reply:
I assume this to not fail, but I read the examples in manual and in case of error it was not calling `regfree`. 
But I think I should explicitly check there if one was compiled successfully free it before throwing, as there might be case when one fails to compile but the other one compiles successfully